addons/sale_coupon/models/sale_coupon_program.py

```diff
@@ -191,18 +191,22 @@ def _compute_program_amount(self, field, currency_to):
 
     @api.model
     def _filter_on_mimimum_amount(self, order):
-        untaxed_amount = order.amount_untaxed
-        tax_amount = order.amount_tax
-
-        # Some lines should not be considered when checking if threshold is met like delivery
-        untaxed_amount -= sum([line.price_subtotal for line in order._get_no_effect_on_threshold_lines()])
-        tax_amount -= sum([line.price_tax for line in order._get_no_effect_on_threshold_lines()])
-
-        return self.filtered(lambda program:
-            program.rule_minimum_amount_tax_inclusion == 'tax_included' and
-            program._compute_program_amount('rule_minimum_amount', order.currency_id) <= untaxed_amount + tax_amount or
-            program.rule_minimum_amount_tax_inclusion == 'tax_excluded' and
-            program._compute_program_amount('rule_minimum_amount', order.currency_id) <= untaxed_amount)
+        no_effect_lines = order._get_no_effect_on_threshold_lines()
+        delivery_lines = order._get_no_effect_delivery_lines()
+        filtered_programs = self.env['sale.coupon.program']
+        order_amount = {
+            'amount_untaxed' : order.amount_untaxed - sum([line.price_subtotal for line in delivery_lines]),
+            'amount_tax' : order.amount_tax - sum([line.price_tax for line in delivery_lines])
+        }
+        for program in self:
+            lines = no_effect_lines.filtered(lambda line: line.product_id == program.discount_line_product_id or line.product_id == program.reward_id.discount_line_product_id)
```

Comment:
Comment 2: given comment 1, we could remove `_get_no_effect_delivery_lines()` method and do the check directly here, as it is not needed for an override and that would avoid the need of another new method.